Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles.py

```diff
@@ -181,21 +181,32 @@ def PtypString8(data_value):
     def PtypString(data_value):
         if data_value:
             try:
+                if USER_ENCODING:
+                    return data_value.decode(USER_ENCODING, errors="ignore")
                 res = chardet.detect(data_value)
                 enc = res['encoding'] or 'ascii'  # in rare cases chardet fails to detect and return None as encoding
                 if enc != 'ascii':
                     if enc.lower() == 'windows-1252' and res['confidence'] < 0.9:
                         demisto.debug('encoding detection confidence below threshold {}, '
                                       'switching encoding to "windows-1250"'.format(res))
-                        enc = 'windows-1250'
-                    data_value = data_value.decode(enc, errors='ignore').replace('\x00', '')
-                elif '\x00' not in data_value:
-                    data_value = data_value.decode("ascii", errors="ignore").replace('\x00', '')
+
+                        enc = DEFAULT_ENCODING if DEFAULT_ENCODING else 'windows-1250'
```

Comment:
Move this above the `demisto.debug` (currently line 190) and use the value of `enc` in the debug message, so it reflects the real encoding you'll use.

Reply:
Agree, will change it accordingly.